io_scene_nif/properties/armature.py

```diff
@@ -47,64 +47,42 @@
 
 
 class BoneProperty(PropertyGroup):
-    @classmethod
-    def register(cls):
-        bpy.types.Bone.niftools = PointerProperty(
-            name='Niftools Bone Property',
-            description='Additional bone properties used by the Nif File Format',
-            type=cls,
-        )
-        # TODO [object][proprerty][flag] This is probably a general object flag which should be available to all generic objects
-        cls.boneflags = IntProperty(
-            name='Bone Flag',
-            default=0
-        )
-        cls.bonepriority = IntProperty(
-            name='Bone Priority',
-            default=0
-        )
-        cls.longname = StringProperty(
-            name='Nif Long Name'
-        )
-
-    @classmethod
-    def unregister(cls):
-        del bpy.types.Bone.niftools
+    # TODO [object][proprerty][flag] This is probably a general object flag which should be available to all generic objects
```

Comment:
Yep. Having generic object flag would remove duplicate code.

Reply:
Yeah, but it turns out we can't actually give bones the object flag, so we have to dupe the code here. :(

Reply:
On the object processing we should be able to pass in either an object or a bone as we would be looking up the same field, "flags"